sec_certs/dataset/cve.py

```diff
@@ -147,9 +147,35 @@ def from_json(cls, input_path: Union[str, Path]):
             dset = json.load(handle, cls=CustomJSONDecoder)
         return dset
 
-    def get_cve_ids_for_cpe_uri(self, cpe_uri: str) -> Optional[Set[str]]:
+    def _get_cve_ids_for_cpe_uri(self, cpe_uri: str) -> Optional[Set[str]]:
         return self.cpe_to_cve_ids_lookup.get(cpe_uri, None)
 
+    def _get_cves_from_exactly_matched_cpes(self, cpe_matches: set[str]) -> set[str]:
+        cve_ids: set[str] = set()
+
+        for cpe_match in cpe_matches:
+            if cve_set := self._get_cve_ids_for_cpe_uri(cpe_match):
+                cve_ids.update(cve_set)
+
+        return cve_ids
```

Comment:
Nitpicking: Can we have it without the for loop? 😆 If we address the note above, then we could write

```python
def _get_cves_from_exactly_matched_cpes(self, cpe_matches: set[str]) -> set[str]:
    return set(itertools.chain.from_iterable([self._get_cve_ids_for_cpe_uri(x) for x in cpe_matches]))
```